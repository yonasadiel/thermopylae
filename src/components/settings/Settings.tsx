import { useState } from 'react';
import ThemeSettings from './Theme';
import QuoteSettings from './Quote';
import './Settings.css';

export interface SettingsProps {
    onClose: () => void
}

const MENU_THEME = 'theme';
const MENU_QUOTE = 'quote';
const MENU_TERMINAL = 'terminal';

export default function Settings(props: SettingsProps) {
    const { onClose } = props;
    const [menu, setMenu] = useState(MENU_THEME);
    return (
        <div className="settings">
            <div className="close-button" onClick={() => onClose()}>
                {/* TODO: make this custom color */}
                <img alt="cross" src="/assets/cross.svg" width="20px" height="20px" />
            </div>
            <div className="navbar">
                <div className={`menu ${menu === MENU_THEME ? 'active' : ''}`} onClick={() => setMenu(MENU_THEME)}>Theme</div>
                <div className={`menu ${menu === MENU_QUOTE ? 'active' : ''}`} onClick={() => setMenu(MENU_QUOTE)}>Quote</div>
                <div className={`menu ${menu === MENU_TERMINAL ? 'active' : ''}`} onClick={() => setMenu(MENU_TERMINAL)}>Terminal</div>
            </div>
            <div className="divider" />
            <div className="settings-content">
                {menu === MENU_THEME && <ThemeSettings />}
                {menu === MENU_QUOTE && <QuoteSettings />}
                {menu === MENU_TERMINAL && <TerminalSettings />}
            </div>
        </div>
    )
}

const TerminalSettings = () => {
    return <div></div>
}
