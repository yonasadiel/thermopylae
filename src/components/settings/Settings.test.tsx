import { test, expect, describe } from '@jest/globals';
import { render } from '@testing-library/react';
import { fn } from 'jest-mock';
import Settings from './Settings';

describe('Settings', () => {
    test('should render correctly', () => {
        const result = render(<Settings onClose={() => { }} />);
        expect(result.baseElement).toMatchSnapshot();
    });
    test('should call on close when closing', () => {
        const onClose = fn();
        const result = render(<Settings onClose={onClose} />);
        const closeButton = result.getByAltText('cross');
        closeButton.click();
        expect(onClose).toHaveBeenCalled();
    });
});