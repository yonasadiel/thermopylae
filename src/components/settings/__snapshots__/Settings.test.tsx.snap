// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Settings should render correctly 1`] = `
<body>
  <div>
    <div
      class="settings"
    >
      <div
        class="close-button"
      >
        <img
          alt="cross"
          height="20px"
          src="/assets/cross.svg"
          width="20px"
        />
      </div>
      <div
        class="navbar"
      >
        <div
          class="menu active"
        >
          Theme
        </div>
        <div
          class="menu "
        >
          Quote
        </div>
        <div
          class="menu "
        >
          Terminal
        </div>
      </div>
      <div
        class="divider"
      />
      <div
        class="settings-content"
      >
        <div
          class="theme-settings"
        >
          <div
            class="input-group"
          >
            <label>
              Style
            </label>
            <p>
              <input
                type="color"
                value="black"
              />
              <span>
                Background Color
              </span>
            </p>
            <p>
              <input
                type="color"
                value="white"
              />
              <span>
                Color
              </span>
            </p>
          </div>
          <div
            class="input-group"
          >
            <label>
              Background Image
            </label>
            <input
              type="file"
              value=""
            />
          </div>
          <div
            class="input-group"
          >
            <label
              for="particles-enabled"
            >
              Particles
            </label>
            <input
              checked=""
              id="particles-enabled"
              type="checkbox"
            />
            <span>
              Enabled
            </span>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
